sharding/nodesCoordinator/indexHashedNodesCoordinatorLite.go

```diff
@@ -0,0 +1,807 @@
+				ShardID:   shardId,
+			}
+		}
+	}
+
+	return publicKeyToValidatorMap
+}
+
+func (ihgs *IndexHashedNodesCoordinatorLite) computeShardForSelfPublicKey(nodesConfig *EpochNodesConfig) (uint32, bool) {
+	pubKey := ihgs.selfPubKey
+	selfShard := ihgs.shardIDAsObserver
+	epNodesConfig, ok := ihgs.nodesConfig[ihgs.currentEpoch]
+	if ok {
+		log.Trace("computeShardForSelfPublicKey found existing config",
+			"shard", epNodesConfig.ShardID,
+		)
+		selfShard = epNodesConfig.ShardID
+	}
+
+	found, shardId := searchInMap(nodesConfig.EligibleMap, pubKey)
+	if found {
+		log.Trace("computeShardForSelfPublicKey found validator in eligible",
+			"epoch", ihgs.currentEpoch,
+			"shard", shardId,
+			"validator PK", pubKey,
+		)
+		return shardId, true
+	}
+
+	found, shardId = searchInMap(nodesConfig.WaitingMap, pubKey)
+	if found {
+		log.Trace("computeShardForSelfPublicKey found validator in waiting",
+			"epoch", ihgs.currentEpoch,
+			"shard", shardId,
+			"validator PK", pubKey,
+		)
+		return shardId, true
+	}
+
+	found, shardId = searchInMap(nodesConfig.LeavingMap, pubKey)
+	if found {
+		log.Trace("computeShardForSelfPublicKey found validator in leaving",
+			"epoch", ihgs.currentEpoch,
+			"shard", shardId,
+			"validator PK", pubKey,
+		)
+		return shardId, true
+	}
+
+	log.Trace("computeShardForSelfPublicKey returned default",
+		"shard", selfShard,
+	)
+	return selfShard, false
+}
+
+// ConsensusGroupSize returns the consensus group size for a specific shard
+func (ihgs *IndexHashedNodesCoordinatorLite) ConsensusGroupSize(
+	shardID uint32,
+) int {
+	if shardID == core.MetachainShardId {
+		return ihgs.metaConsensusGroupSize
+	}
+
+	return ihgs.shardConsensusGroupSize
+}
+
+// GetLastEpochConfig returns the last epoch from nodes config
+func (ihgs *IndexHashedNodesCoordinatorLite) GetLastEpochConfig() uint32 {
+	ihgs.mutNodesConfig.Lock()
+	defer ihgs.mutNodesConfig.Unlock()
+
+	lastEpoch := uint32(0)
+	for epoch := range ihgs.nodesConfig {
+		if lastEpoch < epoch {
+			lastEpoch = epoch
+		}
+	}
+
+	return lastEpoch
+}
+
+// GetNumTotalEligible returns the number of total eligible accross all shards from current setup
+func (ihgs *IndexHashedNodesCoordinatorLite) GetNumTotalEligible() uint64 {
+	return ihgs.numTotalEligible
+}
+
+// GetCurrentEpoch returns current epoch
+func (ihgs *IndexHashedNodesCoordinatorLite) GetCurrentEpoch() uint32 {
+	return ihgs.currentEpoch
+}
+
+// SetCurrentEpoch updates current epoch
+func (ihgs *IndexHashedNodesCoordinatorLite) SetCurrentEpoch(epoch uint32) {
+	ihgs.currentEpoch = epoch
+}
+
+// SetNodesConfig updates nodes config in a concurrent safe way
+func (ihgs *IndexHashedNodesCoordinatorLite) SetNodesConfig(nodesConfig map[uint32]*EpochNodesConfig) {
+	ihgs.mutNodesConfig.Lock()
+	ihgs.nodesConfig = nodesConfig
+	ihgs.mutNodesConfig.Unlock()
+}
+
+// GetNodesConfig returns nodes config for all epochs
+func (ihgs *IndexHashedNodesCoordinatorLite) GetNodesConfig() map[uint32]*EpochNodesConfig {
```

Comment:
Is it safe to return the pointer to internal private member? Why make `nodesConfig` field private in the first place if you have Get/Set with pointer receiver?